fixtures/external-types/lib/src/external-types-lib.udl

```diff
@@ -0,0 +1,16 @@
+namespace external_types_lib {
+    CombinedType get_combined_type(optional CombinedType? cval);
+};
+
+dictionary CrateOneType {
```

Comment:
Hmm if we stick with the UDL approach I wonder if we'll need something to denote that this is indeed external from the current crate (even though it's awesome we can do this now) even if functionally it does nothing. Something like `[ExternalType]` or even `[Crate=<ExternalCrateName>]` feels like it'd help the consumer/viewer quickly understand (imagine if we're using types from 5+ crates the mental juggling would become unwieldy). 

However this just might be me -- open to what other people think

Reply:
This is interesting because Rust doesn't actually need to be told where it is, because it's just a "plain old type" - so it's actually the including Rust code that must have the `use crate_one::CrateOneType` - I was actually surprised this didn't work on main today, but forgot that uniffi in the `lib` crate will try to generate `impl uniffi::RustBufferViaFfi for CrateOneType {` - which fails for the reasons Ben mentions in the adr.

In practice, I doubt that this will end up being a common use-case (ie, that the .udl for a type will be in a different crate than where the type is defined) - I think we'll end up with `crate-one` having its own .udl, and `lib` will want to refer to that via the `typedef` approach we are experimenting with elsewhere. But this is still a good step on the path to that.

Reply:
I'm 50/50 on if it should be tagged in the UDL.  On one hand, it's nice to make it clear where the thing is coming from.  On the other hand, it's kind of redundant.  I went this way because it was just easier to implement, but either is fine with me.